server/app/models/distributed_lock.rb

```diff
@@ -13,17 +15,16 @@ def self.with_lock(name, timeout = 60)
     lock_id = nil
     begin
       if timeout.to_f > 0.0
-        Timeout.timeout(timeout) do
-          sleep 0.01 until lock_id = self.obtain_lock(name)
-        end
+        lock_id = WaitHelper.wait(timeout: timeout) { self.obtain_lock(name) }
```

Comment:
I think it's best to keep discussion of things in the main git `master` PR, unless they're specific to the v1.1 backport.